src/libcollections/string.rs

```diff
@@ -44,7 +78,26 @@ pub struct FromUtf8Error {
     error: Utf8Error,
 }
 
-/// A possible error value from the `String::from_utf16` function.
+/// A possible error value when converting a `String` from a UTF-16 byte slice.
+///
+/// This type is the error type for the [`from_utf16()`] method on [`String`].
+/// It provides no specific details on the kind of error, just that one
+/// occurred.
```

Comment:
This seems a little odd to document like this, the intention here is to have an opaque type that we can then modify in the future to have more internals if necessary, so can this sentence perhaps just be dropped?